llvm/test/Transforms/PGOProfile/thinlto_indirect_call_promotion.ll

```diff
@@ -1,39 +1,45 @@
-; Do setup work for all below tests: generate bitcode and combined index
-; RUN: opt -module-summary %s -o %t.bc
-; RUN: opt -module-summary %p/Inputs/thinlto_indirect_call_promotion.ll -o %t2.bc
+; The raw profiles and reduced IR inputs are generated from Inputs/update_icall_promotion_inputs.sh
+
+; Do setup work for all below tests: annotate value profiles, generate bitcode and combined index.
+; Explicitly turn off ICP pass in Inputs/thinlto_indirect_call_promotion.ll.
+; This way ICP happens in %t.bc after _Z11global_funcv and two indirect callees are imported.
+; RUN: opt -passes=pgo-instr-use -pgo-test-profile-file=%p/Inputs/thinlto_icall_prom.profdata -module-summary %s -o %t.bc
```

Comment:
I think it's a bit cleaner to use [`split-file`](https://llvm.org/docs/TestingGuide.html#extra-files) and have all the `.ll` code in this file instead of putting different files in `Inputs/`. You can even have the `.proftext` file here if that works.

Reply:
Thanks for the pointer to split-file. Applied it for reduced IR. Choose to use raw profiles (see the other comment) per discussion.

Reply:
A common pattern is `rm -rf %t && split-file %s %t`. When dealing with multiple files, it is useful to ensure the temporary directory is cleared.  If specifying `%t/out.bc` is too inconvenient, you can run `rm -rf %t && split-file %s %t && cd %t` and then forget about `%t`.